.github/workflows/on-pull-request.yaml

```diff
@@ -32,6 +32,20 @@ jobs:
         # --frozen-lockfile: doesn't generate a yarn.lock file, fails if an update is needed.
         run: yarn install --frozen-lockfile
 
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v23.2
+        with:
+          files: |
+            src/**/*.ts
+            src/**/*.tsx
+
+      - name: Lint all changed files
+        run: |
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            yarn lint-touched-file $file
+          done
```

Comment:
Please be aware that all these files you run the lint in this step will be lint again in the build step. Please consider decomposing the build step.